tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc

```diff
@@ -515,6 +599,11 @@ StatusOr<bool> CudnnConvAlgorithmPicker::RunOnInstruction(
                       instr->backend_config<CudnnConvBackendConfig>());
   backend_config.set_algorithm(best_algo.conv().algorithm());
   backend_config.set_tensor_ops_enabled(best_algo.conv().tensor_ops_enabled());
+#if TENSORFLOW_USE_ROCM
+  // ROCm: Needs to set scratch size for ROCm stream executor make sure scratch
+  // space for convolution related calls are correctly allocated
+  backend_config.set_scratch_size(best_algo.scratch_bytes());
+#endif
```

Comment:
@timshen91 This is another place where `set_scratch_size()` has to be called inside `rocm` build

Reply:
We already store the scratch bytes at line 521. XLA GPU convolution has two output, the result buffer and the scratch buffer. This is because XLA doesn't quite have a notion for in-out parameter, and we use output parameter for scratch buffer.

Reply:
For clarification, I was actually talking about adding the field in the end of `CudnnConvBackendConfig` [protobuf](https://github.com/tensorflow/tensorflow/blob/ece727fd8c8d278a32357702fddb66caf40d4572/tensorflow/compiler/xla/service/gpu/backend_configs.proto#L43). Which line 521 were you talking about?

Reply:
```
  Shape new_call_shape = ShapeUtil::MakeTupleShape(
      {instr->shape().tuple_shapes(0),
       ShapeUtil::MakeShape(U8, {best_algo.scratch_bytes()})});
```

It stores the scratch size as part of the output tuple shape.